partiql-plan/src/main/java/org/partiql/plan/rex/RexCast.java

```diff
@@ -0,0 +1,71 @@
+package org.partiql.plan.rex;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.plan.Operator;
+import org.partiql.plan.OperatorVisitor;
+import org.partiql.types.PType;
+
+import java.util.List;
+
+/**
+ * Logical cast expression abstract base class.
+ */
+public abstract class RexCast extends RexBase {
+
+    /**
+     * @return new RexCast instance
+     */
+    @NotNull
+    public static RexCast create(@NotNull Rex operand, @NotNull PType target) {
+        return new Impl(operand, target);
+    }
+
+    /**
+     * @return operand rex (operand 0)
+     */
+    @NotNull
+    public abstract Rex getOperand();
+
+    /**
+     * @return target type
+     */
+    @NotNull
+    public abstract PType getTarget();
+
+    @NotNull
+    protected final RexType type() {
+        return RexType.of(getTarget());
+    }
+
+    @Override
+    protected List<Operator> operands() {
+        return List.of();
```

Comment:
Needs operand.

Also, this is interesting regarding the type. If we wanted to add a wrapper PlanNode around PType (for whatever reason), you could add the type to the children. Though, probably not necessary.

Reply:
There's no PlanNode, just operator tree and types aren't operators.